src/Nest/CommonOptions/TimeUnit/Time.cs

```diff
@@ -20,7 +21,7 @@ public class Time : IComparable<Time>, IEquatable<Time>
 		private const double MicrosecondsInAMillisecond = 10;
 
 		private static readonly Regex ExpressionRegex =
-			new Regex(@"^(?<factor>[-+]?\d+(?:\.\d+)?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
+			new Regex(@"^(?<factor>[-+]?[\d\.e\-]+?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
```

Comment:
How about `^(?<factor>[+\-]?(?:\d{1,})(?:\.(?:\d+[e][+\-]?)?\d+)?)[ ]{0,10}(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros)?)$`

`(?:0|[1-9]\d*)` => `(?:\d{1,})` 
leading zeros OK i think
`(?:\.\d*)?(?:[eE][+\-]?\d+)` => `(?:\.(?:\d+[e][+\-]?)?\d+)?)`
doing insensitive matching only `e` should be fine and now with this we do not match on `123e123123` or  `123.e1231`

`\s*` => `[ ]{0,10}` 
only allow 10 spaces so we fail early on `\d<gazillion spaces>...`